File: Application/src/main/java/com/shareyourproxy/api/rx/command/eventcallback/UserMessagesDownloadedEventCallback.java
package com.shareyourproxy.api.rx.command.eventcallback;

import android.app.Notification;
import android.os.Parcel;

import java.util.ArrayList;

/**
 * Created by Evan on 6/18/15.
 */
public class UserMessagesDownloadedEventCallback extends EventCallback {
    private final static java.lang.ClassLoader CL =
        UserMessagesDownloadedEventCallback.class.getClassLoader();
    public static final Creator<UserMessagesDownloadedEventCallback> CREATOR =
        new Creator<UserMessagesDownloadedEventCallback>() {
            @Override
            public UserMessagesDownloadedEventCallback createFromParcel(Parcel in) {
                return new UserMessagesDownloadedEventCallback(
                    (ArrayList<Notification>) in.readValue(CL));
            }

            @Override
            public UserMessagesDownloadedEventCallback[] newArray(int size) {
                return new UserMessagesDownloadedEventCallback[size];
            }
        };
    public final ArrayList<Notification> notifications;

    public UserMessagesDownloadedEventCallback(ArrayList<Notification> notifications) {
        this.notifications = notifications;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(notifications);
    }
}

File: Application/src/main/kotlin/com/shareyourproxy/api/rx/command/eventcallback/UserMessagesDownloadedEventCallback.kt
package com.shareyourproxy.api.rx.command.eventcallback

import android.app.Notification
import android.os.Parcel
import android.os.Parcelable
import java.util.*

/**
 * User messages have been downloaded.
 */
internal final class UserMessagesDownloadedEventCallback(val notifications: ArrayList<Notification>) : EventCallback() {

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(notifications)
    }

    companion object {
        private val CL = UserMessagesDownloadedEventCallback::class.java.classLoader
        @JvmField val CREATOR: Parcelable.Creator<UserMessagesDownloadedEventCallback> = object : Parcelable.Creator<UserMessagesDownloadedEventCallback> {
            @Suppress("UNCHECKED_CAST")
            override fun createFromParcel(parcel: Parcel): UserMessagesDownloadedEventCallback {
                return UserMessagesDownloadedEventCallback(parcel.readValue(CL) as ArrayList<Notification>)
            }

            override fun newArray(size: Int): Array<UserMessagesDownloadedEventCallback?> {
                return arrayOfNulls(size)
            }
        }
    }
}
